go/systemd/systemd_linux.go

```diff
@@ -9,8 +9,13 @@ import (
 
 	sdActivation "github.com/coreos/go-systemd/activation"
 	sdDaemon "github.com/coreos/go-systemd/daemon"
+	sdUtil "github.com/coreos/go-systemd/util"
 )
 
+func IsRunningSystemd() bool {
+	return sdUtil.IsRunningSystemd()
```

Comment:
can you explain how this works?

Reply:
This ends up just making a [simple stat call](https://github.com/coreos/go-systemd/blob/master/util/util.go#L72-L78):

```go
func IsRunningSystemd() bool {
	fi, err := os.Lstat("/run/systemd/system")
	if err != nil {
		return false
	}
	return fi.IsDir()
}
```

Reply:
I should clarify that this is checking whether systemd is running as root. It's not directly looking for the current user's user-level systemd process. So it's possible that we'll run into corner cases where we try the systemd route, but it fails. The main one I can think of is if I do something like `sudo su other_user` and then `keybase ctl start` in that shell. I think the local session won't be set up in that case (or even if it is, I won't have the relevant `DBUS` variables to talk to it), so startup will fail. That said, firing off demons as other user accounts was always kind of weird. I'm not sure how much we want to handle these cases?

Reply:
Actually I might change this to our own function that does the same thing, but for the a user-specific folder in the `XDG_RUNTIME_DIR`.